provisioning/src/main/java/com/espressif/provisioning/ESPDevice.java

```diff
@@ -481,6 +483,93 @@ public void onFailure(Exception e) {
         }
     }
 
+    /**
+     * Send Enterprise Wi-Fi credentials to device for provisioning.
+     *
+     * @param username Username of the Wi-Fi which is to be configure in device.
+     * @param password Password of the Wi-Fi which is to be configure in device.
+     * @param listener Listener to get success and failure.
+     */
+    public void setEnterpriseCredentials(final String username, final String password, final ResponseListener listener) {
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject
+                    .put("device_token", deviceAccessToken)
+                    .put("username", username)
+                    .put("password", password);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        sendJSONObjectToCustomEndPoint("set-wpa2-ent-creds", jsonObject, listener);
+    }
+
+    /**
+     * Send Enterprise Wi-Fi certificate to device for provisioning.
+     *
+     * @param certificate Certificate of the Wi-Fi which is to be configure in device.
+     * @param listener    Listener to get success and failure.
+     */
+    public void setEnterpriseCertificate(final String certificate, final ResponseListener listener) {
+        setEnterpriseCertificate(certificate, 0, listener);
+    }
+
+    /**
+     * Send Enterprise Wi-Fi certificate to device for provisioning.
+     *
+     * @param certificate Certificate of the Wi-Fi which is to be configure in device.
+     * @param sequence    sequence of the certificate.
+     * @param listener    Listener to get success and failure.
+     */
+    private void setEnterpriseCertificate(final String certificate, final int sequence, final ResponseListener listener) {
+        int beginIndex = sequence * 256;
+        int endIndex = (sequence + 1) * 256;
```

Comment:
Can you define and use private constants for:
"device_token"
"username"
"password"
256, 
etc?

Reply:
"seq", "chunk", "last", etc. too

Reply:
Same for endpoints names too please. Maybe in the same place that ESP defines them.